code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -231,9 +231,14 @@
 			var/amount_per_pill = reagents.total_volume/count
 			if (amount_per_pill > 60) amount_per_pill = 60
 
-			var/name = sanitizeSafe(input(usr,"Name:","Name your pill!","[reagents.get_master_reagent_name()] ([amount_per_pill] units)") as null|text, MAX_NAME_LEN)
-			if(!name) //Blank name (sanitized to nothing, or left empty) or cancel
-				return
+			if(condi)//For the condimaster
+				var/name = sanitizeSafe(input(usr,"Name:","Name your cube!","[reagents.get_master_reagent_name()] ([amount_per_pill] units)") as null|text, MAX_NAME_LEN)
+				if(!name) //Repeated because i suck

```

Comment:
You can pull both of these `if(!name)`s out of the if else block and put it after the else. The code needs to run regardless of the outcome of the if, so it shouldn't go in the if

Reply:
If there is no name, a return occurs and no pill is produced.